src/main/scala/dpla/ingestion3/enrichments/SpatialEnrichment.scala

```diff
@@ -0,0 +1,275 @@
+    * @param filterWoeType  What-on-Earth Type number
+    * @return
+    */
+  private def parentFeatureName(json: JValue, filterWoeType: Int):
+      Option[String] = {
+    val featNames: List[String] = for {
+      JArray(parents) <- json \\ "parents"
+      JObject(parent) <- parents
+      JField("name", JString(name)) <- parent
+      JField("woeType", JInt(woeType)) <- parent
+      if woeType == filterWoeType
+    } yield name
+    if (featNames.nonEmpty) Option(featNames.head)
+    else None
+  }
+
+  /**
+    * Return the city / town name of the Place.
+    *
+    * Note that this is not handled by `parentFeatureName` because the City name is
+    * going to be the name of the place, City / Town being the smallest level
+    * of detail reported by Twofishes. It's not going to be a parent feature.
+    *
+    * @param j  Twofishes response body JValue
+    * @return   The city / town name
+    */
+  private def city(j: JValue): Option[String] = {
+    val feature = ((j \ "interpretations")(0) \ "feature").extract[Feature]
+    if (feature.woeType == 7) {
+      Option(feature.name)
+    } else {
+      None
+    }
+  }
+
+  /* Similar to `city`, return the other parent feature names ... */
+  private def county(j: JValue): Option[String] = {
+    if (woeType(j) == Some(9)) shortName(j)
+    else parentFeatureName(j, 9)
+  }
+  private def state(j: JValue): Option[String] = {
+    if (woeType(j) == Some(8)) shortName(j)
+    else parentFeatureName(j, 8)
+  }
+  private def country(j: JValue): Option[String] = {
+    if (woeType(j) == Some(12)) shortName(j)
+    else parentFeatureName(j, 12)
+  }
+
+  // Region is implemented inconsistently in the Ingestion1 Twofishes enricher,
+  // and Twofishes does not return it consistently. If you look up "United
+  // States" it will provide the continent, "North America", but if you look up
+  // a city in the United States it will not return the continent.  Filling in
+  // a region value will provide misleading searches.
+  // private def region(j: JValue): String = ???
+
+  /**
+    * Return a single String of the Place's coordinates, joined by ","
+    *
+    * @param j  Twofishes response JValue
+    * @return   The coordinates as at "lat, lng" string
+    */
+  private def coordinates(j: JValue): Option[String] = {
+    val c = ((((j \ "interpretations")(0) \ "feature") \ "geometry") \
+      "center").extract[Coords]
+    Option(c.lat.toString + "," + c.lng.toString)
+  }
+
+  /**
+    * Return the place name or None if it's the place's coordinates
+    *
+    * If the place's name field has been used to convey coordinates, we do want
+    * to override this when we fill it in.  Otherwise, we don't want to clobber
+    * it if it's already defined.
+    *
+    * @param nameCandidate
+    * @param j
+    */
+  private def nameUnlessCoords(nameCandidate: Option[String], j: JValue):
+      Option[String] = {
+    nameCandidate match {
+      case s if s.isDefined => {
```

Comment:
Here I would say `case Some(value)` and let the pattern match destructure the Option for you.